### PR TITLE
Add keywords.deleter: Fixes #470

### LINKED ISSUE #470: Add keywords.deleter for ImageFileCollection
Currently the summary of a ImageFileCollection can be wiped by setting the keywords to None, however using `del xxx.keywords` fails because there is no deleter assigned to the property.

I think it would be a good idea to have that deleter for the keywords (and maybe deprecate the option to wipe the keywords by setting `xxx.keywords = None`, I mean it's also possible to almost wipe them when one uses `xxx.keywords = []`).

Also currently the `keywords.setter` has a default value of `None` for the `keywords` which makes no sense (except one accesses the `__class__.keywords.fset`-method directly).

### PR DESCRIPTION
**Pull request by janga1997**
Please have a look at the following list and replace the "[ ]" with a "[x]" if
the answer to this question is yes.

For new functionality:

- [x] Did you add an entry to the "Changes.rst" file?
- [x] Did you include a meaningful docstring with Parameters, Returns and Examples?
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [x] Did you include tests for the new functionality?
- [x] Does this PR add, rename, move or remove any existing functions or parameters?

### CODE REVIEW

**Comment by MSeifert04** [comment]
Needs a test and a changelog entry. Otherwise this looks good 👍

File: CHANGES.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ Other Changes and Additions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 - removed ability to set ccdata.unit to None [#451]
+- Add keywords.deleter method to ImageFileCollection [#474]
```
Comment (MSeifert04) [inline CHANGES.rst]: Could you add a blank line between the items? Also this is a "new feature".

File: ccdproc/tests/test_image_collection.py [hunk 2]
```diff
@@ -101,6 +101,13 @@ def test_filenames_are_set_properly(self, triage_setup):
             location=triage_setup.test_dir, filenames=fn, keywords=['filter'])
         assert img_collection.files == [fn]
 
+    def test_keywords_deleter(self, triage_setup):
+        ic = image_collection.ImageFileCollection(triage_setup.test_dir,
+                                                  keywords='*')
+
+        del ic.keywords
```
Comment (MSeifert04) [inline ccdproc/tests/test_image_collection.py]: to make the check more robust please also check that the `keywords` before deleting them are not empty.

**Comment by MSeifert04** [comment]
Apart from the inline-comments could you add a 

```
.. versionchanged:: 1.3
   Added ``deleter`` for ``keywords`` property.
```

in the docstring of the ``keywords`` property.

File: ccdproc/image_collection.py [hunk 3]
```diff
@@ -211,6 +211,10 @@ def keywords(self, keywords=None):
 
     @keywords.deleter
     def keywords(self):
+        """
+        versionchanged:: 1.3
```
Comment (MSeifert04) [inline ccdproc/image_collection.py]: Has to be in the property not in the deleter. Also indentation and the `..` are important!

**Comment by MSeifert04** [comment]
If you have time could you build the documentation and check that the ``keywords`` property renders correctly?

File: CHANGES.rst [hunk 4]
```diff
@@ -5,13 +5,13 @@
...
 ^^^^^^^^^^^^
 
+- Add keywords.deleter method to ImageFileCollection [#474]
 
 Other Changes and Additions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 - removed ability to set ccdata.unit to None [#451]
 
-
```
Comment (MSeifert04) [inline CHANGES.rst]: Could you add the blank line again?

File: CHANGES.rst [hunk 5]
```diff
@@ -5,13 +5,13 @@
 New Features
 ^^^^^^^^^^^^
 
+- Add keywords.deleter method to ImageFileCollection [#474]
```
Comment (MSeifert04) [inline CHANGES.rst]: Could you add a blank line after the item here? Makes it easier to distinguish sections.

**Comment by janga1997** [comment]
@MSeifert04 I would , but its 5:30 am here!
I will do that and squash the commits later. BTW, could I have a final word on #475  and #463 ?

**Comment by MSeifert04** [comment]
@janga1997 no rush. Just let me know if you encounter problems :)

 I'll have a look at the other two pull requests.

**Comment by mwcraig** [comment]
> @mwcraig @crawfordsm do you want to have a look?

Looks good to me!

@janga1997 a squash would be great, no rush at all (there is no requirement that a PR be merged, just opened and worked on, and you already have several 😃 )

I've left comments on the other two PRs too.

**Comment by MSeifert04** [comment]
I restarted the failed test. Otherwise LGTM 👍

**Comment by mwcraig** [comment]
Merging, thanks @janga1997 for the PR, and @MSeifert04 for the review and coaching!

**Comment by janga1997** [comment]
This was more @MSeifert04 's PR than me. I don't think I have seen another organisation that is this thorough with their PRs.

**Comment by mwcraig** [comment]
We do have a pretty high ratio of discussion comments per contributed line of code. The winner by that metric may be https://github.com/astropy/astropy/pull/3333...

**Comment by bsipocz** [comment]
@mwcraig - You've clearly missed out on https://github.com/astropy/astropy/pull/5759

**Comment by MSeifert04** [comment]
am I guilty of bikeshedding?

**Comment by bsipocz** [comment]
We all are. But the result is a nicer and cleaner codebase, so I think it's still on an acceptable level on a cost/benefit scale.

**Comment by janga1997** [comment]
@MSeifert04 I had to google that term, but definitely not. My first drafts may work, but I think the edits you guys request gives it the ship quality.

**Comment by mwcraig** [comment]
This one was also mixed with some git-fu too...

**Pull request merged**